src/scripts/main.js

```diff
@@ -1,3 +1,49 @@
 'use strict';
 
-// write code here
+const sortEmployeesBySalary = function (selector) {
+  // Получаем все элементы списка
+  const employeeItems = document.querySelectorAll(selector);
+
+  // Преобразуем NodeList в массив и сортируем его по зарплате
+  const sortedEmployees = Array.from(employeeItems).sort((a, b) => {
+    // Извлекаем зарплату из атрибутов данных и преобразуем в число
+    const salaryA = parseInt(a.dataset.salary.replace(/[$,]/g, ''), 10);
+    // Удаляем $ и запятые
+    const salaryB = parseInt(b.dataset.salary.replace(/[$,]/g, ''), 10);
+
+    return salaryB - salaryA; // Сортируем по убыванию
+  });
+
+  // Очищаем существующий список и добавляем отсортированные элементы
+  const employeeList = document.querySelector('ul');
+
+  employeeList.innerHTML = ''; // Очищаем существующий список
```

Comment:
Performance Issue: Clearing the innerHTML of an element is a heavy operation and can cause performance issues if done frequently. Since you are appending sorted children afterwards, there is no need to clear the innerHTML. The appendChild method will automatically remove the element from its initial position and append it to the new position.